src/main/java/com/aws/greengrass/builtin/services/mqttproxy/MqttProxyIPCAgent.java

```diff
@@ -212,21 +212,19 @@ public CompletableFuture<SubscribeToIoTCoreResponse> handleRequestAsync(Subscrib
                                 .log("Unable to subscribe to topic");
                         throw new ServiceError(String.format("Subscribe to topic %s failed with error %s", topic, t));
                     }).thenApply((i) -> {
-                        if (i != null) {
-                            int rc = i.getReasonCode();
-                            if (rc > 2) {
-                                String rcString = SubAckPacket.SubAckReasonCode.UNSPECIFIED_ERROR.name();
-                                try {
-                                    rcString = SubAckPacket.SubAckReasonCode.getEnumValueFromInteger(rc).name();
-                                } catch (RuntimeException ignored) {
-                                }
-
-                                throw new ServiceError(
-                                        String.format("Subscribe to topic %s failed with error %s", topic,
-                                                rcString))
-                                        .withContext(Utils.immutableMap("reasonString", i.getReasonString(),
-                                                "reasonCode", i.getReasonCode()));
+                        if (i != null && !i.isSuccessful()) {
+                            String rcString = SubAckPacket.SubAckReasonCode.UNSPECIFIED_ERROR.name();
+                            try {
+                                rcString =
+                                        SubAckPacket.SubAckReasonCode.getEnumValueFromInteger(i.getReasonCode()).name();
+                            } catch (RuntimeException ignored) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception has been ignored. The catch-block hides information about the stack trace. Not logging or forwarding the stack trace of an exception can mask unexpected errors and complicated debugging.

**Fix**: Use a catch block to handle the exception and write a log message declaring that the exception happened. 
[Learn more.](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)

Suggested remediation:
Handle all exceptions by creating a log for each exception using the message thrown.
```
@@ -220,2 +220,3 @@
                             } catch (RuntimeException ignored) {
+                                log.error(&#34;RuntimeException is caught.&#34;, ignored);
                             }

```